src/iexamine.cpp

```diff
@@ -3044,6 +3045,22 @@ void iexamine::water_source(player &p, const tripoint &examp)
     g->handle_liquid( water, nullptr, 0, &examp );
 }
 
+void iexamine::gather_snow(player &p, const tripoint &examp)
+{
+    if(!query_yn(_("Gather snow?"))) {
+        none( p, examp );
+        return;
+    }
+    if( p.can_pickWeight( item( "snow" ), true ) && p.can_pickVolume( item( "snow" ), true ) ){
+        p.i_add( item( "snow" ) );
+        p.add_msg_if_player( _( "You gathered some snow" ) );
+    } else {
+        g->m.add_item_or_charges( p.pos(), item( "snow" ) );
```

Comment:
`item` gets constructed four times (isn't quite cheap) from the same string literal. I'd extract a temporary for both deduplication and performance.